src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/UnitTests/Controllers/HomeControllerTests.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            this._homeServiceMock = new Mock<IHomeService>();
+        }
+
+        [Fact]
+        public void Index_CurrentTestDtoIsNull_ReturnsEmptyViewResult()
+        {
+            // Arrange
+            this._homeServiceMock.Setup(x => x.GetCurrentTests()).Returns((List<TestDto>)null);
+            using var controller = new HomeController(this._homeServiceMock.Object);
+
+            // Act
+            var result = controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.Model);
+        }
+
+        [Fact]
+        public void Index_UserIsNull_ReturnsAViewResutWithListOfTestDto()
+        {
+            // Arrange
+            var currentTestList = new List<TestDto>()
+            {
+                GetTestDto()
+            };
+            this._homeServiceMock.Setup(x => x.GetCurrentTests()).Returns(currentTestList);
+            using var controller = new HomeController(this._homeServiceMock.Object);
+            controller.ControllerContext = new ControllerContext()
+            {
+                HttpContext = new DefaultHttpContext() { User = null }
+            };
+
+            // Act
+            var result = controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.IsType<List<TestDto>>(viewResult.ViewData.Model);
+        }
+
+        [Fact]
+        public void Index_ReturnsCorrectAnswersForUser()
+        {
+            // Arrange
+            var httpContext = new DefaultHttpContext();
+            var user = GetUser();
+            var userId = user.FindFirstValue(ClaimTypes.NameIdentifier);
+            httpContext.HttpContext.User = user;
+
+            var currentTestList = new List<TestDto>
+            {
+                GetTestDto()
+            };
+
+            this._homeServiceMock.Setup(x => x.GetCurrentTests()).Returns(currentTestList);
+            using var controller = new HomeController(this._homeServiceMock.Object)
+            {
+                ControllerContext = new ControllerContext()
+                {
+                    HttpContext = httpContext
+                }
+            };
+
+            // Act
+            var result = controller.Index();
+
+            // Assert
+            this._homeServiceMock.Verify(x => x.GetCorrectAnswersCountForUser(userId), Times.Once);
+        }
+
+        [Fact]
+        public void CheckTestStatus_ReturnsContetResultWithTestStatus()
+        {
+            // Arrange
+            var test = GetTest(TestStatus.Started);
+            this._homeServiceMock.Setup(x => x.CheckTestStatus(It.IsAny<int>())).Returns(It.IsAny<string>());
+            using var controller = new HomeController(this._homeServiceMock.Object);
+
+            // Act
+            var result = controller.CheckTestStatus(test.Id);
+
+            // Assert
+            var viewResult = Assert.IsType<ContentResult>(result);
```

Comment:
why do we need this variable?

probably we need sth more to check, not only type

Reply:
I am not sure what else coud we check. I added  this._homeServiceMock.Verify(x => x.CheckTestStatus(test.Id), Times.Once);

Reply:
We can verify the end result of CheckTestStatus action which will be ContentResult. We can set up stub to return certain string and then do something like this `Assert.Equal(contentResult.Content, testStatus);`. That way we won't verify collaboration with a stub, because it doesn’t matter how the SUT generates the end result, as long as that result is correct.

Reply:
@Mejtika correct, commited update